### DISCUSSION TITLE
Authentication error when using dictGet function

### DISCUSSION THREAD

**Discussion by maksspace** [ClickHouse/ClickHouse#51365] [Q&A] [answered]
Hi, I have table `geoip` and I created dict `ip_trie` from it (https://clickhouse.com/blog/geolocating-ips-in-clickhouse-and-grafana).
Im using default user with password set in the default-password.xml.

But when I'm trying to select data from it, it always triggers an authentication error:

```sql
create dictionary ip_trie (
   cidr String,
   latitude Float64,
   longitude Float64,
   country_code String
) 
primary key cidr
source(clickhouse(table ‘geoip’))
layout(ip_trie)
lifetime(3600);
```

```sql
SELECT *
FROM ip_trie
LIMIT 20
```

```
Received exception from server (version 23.5.3):
Code: 516. DB::Exception: Received from localhost:9000. DB::Exception: default: Authentication failed: password is incorrect, or there is no user with such name.

If you have installed ClickHouse and forgot password you can reset it in the configuration file.
The password for default user is typically located at /etc/clickhouse-server/users.d/default-password.xml
and deleting this file will reset the password.
See also /etc/clickhouse-server/users.xml on the server where ClickHouse is installed.

. (AUTHENTICATION_FAILED)
```

**Comment by den-crane** [comment, marked as answer]
>Im using default user with password set in the default-password.xml.

Dictionary should use the exact same password.
If you set a password `xxxx` to default user.
Dictionary must have `password 'xxxx'` section.

**Reply by Golem765** [reply]
Thanks @den-crane, I just want to clarify to someone stumbling on this answer the same way I did - you have to specify the password in the `source` section so that it would be `source(clickhouse(table ‘geoip’ password 'xxxx'))`

**Comment by ivan-klass** [comment]
This is extremely annoying. 
Does it mean that we have inject the password into migrations?
Also, everything works fine without `password` section in the cloud, but seems like it doesn't work with local docker distribution.

**Reply by den-crane** [reply]
https://clickhouse.com/docs/en/operations/named-collections#example-of-using-named-collections-with-a-dictionary-with-source-clickhouse

https://clickhouse.com/docs/en/sql-reference/statements/create/named-collection

**Reply by AlexKichkailo** [reply]
agree with @ivan-klass, named collection is not a good solution. From the UX perspective, dictionaries that point to local CH instance should not require user credentials. Specifically, dictionaries with DIRECT layout should reuse the existing user credentials.

**Reply by grocon** [reply]
agreed with Alex. By the way, clickhouse cloud doesn't support named collections.

**Reply by cpassnat** [reply]
Yes, access to local dictionary should not require a password.
BTW from the documentation it's not clear which user password should be used and what kind of permissions shoud have this user.

**Reply by KIC** [reply]
this is really extremely annoying!! please fix this :pray:
